src/drive/web/modules/views/Sharings/FilesViewerSharings.jsx

```diff
@@ -0,0 +1,50 @@
+import React from 'react'
+import { useQuery } from 'cozy-client'
+import { useRouter } from 'drive/lib/RouterContext'
+import { buildSharingsQuery } from 'drive/web/modules/queries'
+import { getCurrentFileId } from 'drive/web/modules/selectors'
+import { connect } from 'react-redux'
+import FilesViewer, {
+  FilesViewerLoading
+} from 'drive/web/modules/viewer/FilesViewerV2'
+import SharedDocuments from 'cozy-sharing/dist/components/SharedDocuments'
+
+const FilesViewerWithQuery = ({ sharedDocumentIds, ...props }) => {
+  const filesQuery = buildSharingsQuery(sharedDocumentIds)
```

Comment:
I am wondering if the query or its results should not come directly from the parent since they have a clear parent/child relationship and since they use the same query. Wouldn't it be simpler to use directly the raw FilesViewer in the SharingsView ?

Reply:
I wonder about this too, but we also want the viewer to have it's own URL so we have to rely on react-router to a degree for this. It's a good point but I'd rather tackle this in another PR when we decide to do the change.

Reply:
OK, no problem.